server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java

```diff
@@ -5827,4 +5830,28 @@ public void mimeMessageIdShouldBePreservedWhenSending() {
         assertThat(receivedMimeMessageId).isEqualTo(creationMimeMessageId);
     }
 
+    @Test
+    public void setMessagesShouldWorkForHugeNumberOfEmailsToTrash() throws Exception {
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        String mailIds = IntStream
+            .rangeClosed(1, NUMBER_OF_MAIL_TO_CREATE)
+            .mapToObj(Throwing.intFunction((i) ->
+                mailboxProbe
+                    .appendMessage(USERNAME, MailboxPath.forUser(USERNAME, DefaultMailboxes.TRASH),
```

Comment:
I'd advocate to use the AppendCommand API in MailboxProbeImpl. 

That way we would avoid maaany noise here.

(Use `ComposedMessageId appendMessage(String username, MailboxPath mailboxPath, MessageManager.AppendCommand appendCommand)` instead...)

Reply:
Extract the mailbox path to a constant for readability...